packages/css/src/generate-compression-map.ts

```diff
@@ -45,23 +47,56 @@ export const generateCompressionMap = (
   result.css;
 
   // Remove duplicates
-  classNamesToCompress = Array.from(new Set(classNamesToCompress));
+  atomicClassNames = Array.from(new Set(atomicClassNames));
+  nonAtomicClassNames = Array.from(new Set(nonAtomicClassNames));
 
-  // Check if class name to compress already exists in oldClassNameCompressionMap
-  // If yes, re-use the compressed class name
+  // `oldClassNameCompressionMap` is used to ensure class names are consistent between builds.
+  // It means if `aaaabbbb` gets compressed to `a` previously, it needs to be compressed to `a` again.
+  // If a compressed name exists in both stylesheet and old compression map, we assume it's a previously compressed class name, and keep it.
   if (oldClassNameCompressionMap) {
-    classNamesToCompress = classNamesToCompress.filter((className) => {
+    atomicClassNames = atomicClassNames.filter((className) => {
       if (oldClassNameCompressionMap[className]) {
         reservedClassNames.push(oldClassNameCompressionMap[className]);
         classNameCompressionMap[className] = oldClassNameCompressionMap[className];
         return false;
       }
       return true;
     });
+
+    if (Object.keys(classNameCompressionMap).length) {
+      // We've found class names which exist in old compression map but not in the stylesheet.
+      // Theoretically this should not happen. We want to warn the developers so they can investigate it. There is likely some race condition in the builds.
+      console.warn(
+        `${JSON.stringify(
```

Comment:
maybe I'm misunderstanding the code here, but are you logging the right reference?

Reply:
also couldn't this happen if someone deletes a style? Doesn't seem like a bug if that happens

Reply:
> also couldn't this happen if someone deletes a style? Doesn't seem like a bug if that happens

The input stylesheet is `compiled-css.css` from Webpacking so it's likely to include compressed class names already. This check is checking whether a style is supposed to be compressed (exists in old compression map) but isn't compressed in the stylesheet. Let me give an example to explain it.

### First pass
#### stylesheet
```
.aaaabbbb {
    color: red
}
.aaaacccc {
    color: green 
}
```
#### old compression map
undefined
#### generated compression map
```
{
  "aaaabbbb": "a",
  "aaaacccc": "c"
}
```
### second pass:
#### stylesheet
Note `.a` is compressed from `.aaaabbbb`
```
.a {
    color: red
}
.aaaacccc {
    color: green 
}
```
#### old compression map
```
{
  "aaaabbbb": "a",
  "aaaacccc": "c"
}
```
We found `.aaaacccc` exists in the old compression map, which means it should become ".c" instead. The code can handle this, but we want to warn developers that something is not right. At this point of time, `classNameCompressionMap` should be empty (means no unexpectedly uncompressed class names). The cause of this (that I can think of) is some sort of race condition in builds. e.g. multiple generateCompressionMap are running in parallel.

I feel my brain capacity reached to a limit when thinking about different cases 🤯. Does it make any sense?

Reply:
right sounds good, maybe it would be good to comment on the different map variables to describe what they are for code readability. The comments for the errors are good but the context for the variable is missing